File: Algorithms/Greedy/MaxDotProduct.py

def get_max_dot_product(a, b):

    a.sort()
    b.sort()

    return sum( [ a[i] * b[i] for i in range(len(a)) ] )

def main():

    a = [1, 3, -5]

    b = [-2, 4, 1]

    print(get_max_dot_product(a, b))

if __name__ == '__main__':
    main()

File: Algorithms/Greedy/MaxDotProduct.java

import java.util.*;

public class MaxDotProduct {

    public static int getMaxDotProduct(int[] a, int[] b) {

        Arrays.sort(a);
        Arrays.sort(b);

        int sum = 0;

        for (int i = 0; i < a.length; i++) {

            sum += a[i] * b[i];

        }

        return sum;

    }

    public static void main(String[] args) {

        int[] a = new int[]{1, 3, -5};

        int[] b = new int[]{-2, 4, 1};

        System.out.println(getMaxDotProduct(a, b));

    }

}
